src/modules/referrals/reducer.spec.ts

```diff
@@ -0,0 +1,189 @@
+
+      expect(newState.loading).toContainEqual(action)
+      expect(newState.error).toBeNull()
+    })
+  })
+
+  describe('when handling fetchReferralsSuccess', () => {
+    it('should update the data and clear loading state', () => {
+      const mockData: ReferralProgressResponse = {
+        invitedUsersAccepted: 5,
+        invitedUsersAcceptedViewed: 3
+      }
+
+      const state: ReferralsState = {
+        ...initialState,
+        loading: [fetchReferralsRequest() as AnyAction]
+      }
+
+      const action = fetchReferralsSuccess(mockData)
+      const newState = referralsReducer(state, action)
+
+      expect(newState.data.invitedUsersAccepted).toBe(5)
+      expect(newState.data.invitedUsersAcceptedViewed).toBe(3)
+      expect(newState.loading).not.toContainEqual(fetchReferralsRequest())
+    })
+
+    it('should handle zero values correctly', () => {
+      const mockData: ReferralProgressResponse = {
+        invitedUsersAccepted: 0,
+        invitedUsersAcceptedViewed: 0
+      }
+
+      const action = fetchReferralsSuccess(mockData)
+      const newState = referralsReducer(initialState, action)
+
+      expect(newState.data.invitedUsersAccepted).toBe(0)
+      expect(newState.data.invitedUsersAcceptedViewed).toBe(0)
+    })
+
+    it('should handle large values correctly', () => {
+      const mockData: ReferralProgressResponse = {
+        invitedUsersAccepted: 999999,
+        invitedUsersAcceptedViewed: 888888
+      }
+
+      const action = fetchReferralsSuccess(mockData)
+      const newState = referralsReducer(initialState, action)
+
+      expect(newState.data.invitedUsersAccepted).toBe(999999)
+      expect(newState.data.invitedUsersAcceptedViewed).toBe(888888)
+    })
+  })
+
+  describe('when handling fetchReferralsFailure', () => {
+    it('should set error message and clear loading state', () => {
+      const errorMessage = 'Failed to fetch referrals'
+      const state: ReferralsState = {
+        ...initialState,
+        loading: [fetchReferralsRequest() as AnyAction]
+      }
+
+      const action = fetchReferralsFailure(errorMessage)
+      const newState = referralsReducer(state, action)
+
+      expect(newState.error).toBe(errorMessage)
+      expect(newState.loading).not.toContainEqual(fetchReferralsRequest())
+    })
+
+    it('should handle empty error message', () => {
+      const action = fetchReferralsFailure('')
+      const newState = referralsReducer(initialState, action)
+
+      expect(newState.error).toBe('')
+    })
+
+    it('should handle long error messages', () => {
+      const longErrorMessage = 'A'.repeat(1000)
+      const action = fetchReferralsFailure(longErrorMessage)
+      const newState = referralsReducer(initialState, action)
+
+      expect(newState.error).toBe(longErrorMessage)
+    })
+  })
+
+  describe('when handling multiple actions in sequence', () => {
+    it('should handle request -> success flow correctly', () => {
```

Comment:
These tests here are great! But they need to follow our testing standard. For example, in this case, we need to build a state where the request is first send and then child contexts for each of the cases, for example:
```
describe('when handling a fetch referral request', () => {
  beforeEach(() => {
    state = referralsReducer(initialState, fetchReferralsRequest())
  })
  ...
  describe('and handling a fetch referral success', () => {
    beforeEach(() => {
      state = referralsReducer(initialState, fetchReferralSuccess({...}))
    })

    it('should return an state where ....')
  })
})
```